## Examining Student and Online Perceptions of LSE

### Part 1: Introduction

Before arriving at LSE, many students likely hear of the university's low student satisfaction rates. These impressions often gain traction on online forums such as Reddit, where anecdotal accounts tend to amplify perceptions of dissatisfaction. Additionally, multiple university ranking websites also present LSE as having a comparatively low student satisfaction rate. Therefore, it is worth exploring this topic, to understand how "severe" the problem is and whether the data fully aligns with the online perception of the university.

We can explore the statement above through a series of focused questions:
- How does LSE student satisfaction compare to other universities?
- How does student satisfaction within LSE vary by degree?
- How has LSE student satisfaction evolved over the last few years (2020-2023)?
- How is LSE perceived by people online?

It is worth considering what is meant by student satisfaction. Many data sources such as the National Student Survey break down student's views into categories such as teaching, course content, course organisation and assessment. Meanwhile, online the focus tends to be on social opportunities, as well as societies. As "student satisfaction" is multifaceted we will be breaking down each question into the specific type of student satisfaction we are discussing, to avoid confusion. The NSS is generally better suited towards 'academic' related definitions of student satisfaction, such as student's views on the quality of the course, the teaching etc. whilst Reddit and online platforms likely better represent student's thoughts on their social lives at certain universities.




### Part 2: Data acquisition
- Go to 'Data_acquisition.ipynb' file

### Part 3: Data Preparation and Exploration

In [3]:
import pandas as pd
import pickle

### NSS data

We can answer many of the questions through the NSS. The NSS questions students on multiple different categories, such as teaching quality, course quality and assessment quality. Most websites ignore this distinction and simply focus on one of these metrics such as teaching quality, however, we wanted to make the distinction between categories clearer. As there are multiple categories, we create multiple dataframes so that visualisation is easier. Another important thing to consider is the response to population size; if it is high it means the measure of student satisfaction is more likely to accurately represent the real student body, compared to if it is low, when the results are less robust. This is the reason why many online displays of these metrics leave out universities like the University of Oxford, where a hefty proportion of the students boycott the NSS.

We also wanted to display the variance between student's opinions on these measures through the standard deviation score.

In [4]:
file_path = './data/univ_df.pkl'

# Load the dictionary from the pickle file
with open(file_path, 'rb') as pickle_file:
    univ_df = pickle.load(pickle_file)

#refresher of the universities surveyed
for key in univ_df.keys():
    print(key)

LSE
Oxford
UCL
Birmingham
Edinburgh
Glasgow
Imperial
KCL
Manchester
Norwich
Strathclyde
Warwick


Below, we create dataframes representing the differences in student opinions on teaching quality, course quality, assessment quality, support quality, as well as course organisation quality. We considered aggregating every metric into one 'overall satisfaction' score, however, feel that would not take into account the fact that some may view certain criterion as deserving more weight; for example, learning opportunities may be valued much more than assessment quality. These themes are represented on the NSS and are calculated through averaging student responses to questions that can be grouped into one larger theme.


In [101]:
teach_dict={}
learn_dict={}
assess_dict={}
support_dict={}
organisation_dict={}
resource_dict={}
voice_dict={}

for name,uni_df in univ_df.items():
    
    #We measure 'All Subjects' at this point as we don't want to differentiate by Subject
    #The last 7 rows of the dataframe refer to the Themes we are interested in
    df=uni_df[(uni_df["Level of study"]=='All undergraduates') & (uni_df["Subject level"]=="All subjects")& (uni_df['Level of study']=='All undergraduates')][-7:]
    
    #Teaching Scores
    positivity=df.iloc[0]['Positivity measure (%)']
    sd=df.iloc[0]['Standard deviation']
    response_ratio=round(df.iloc[0]['Responses']/df.iloc[0]['Population'],2)
    teach_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}
    
    #Learning Opportunities
    positivity=df.iloc[1]['Positivity measure (%)']
    sd=df.iloc[1]['Standard deviation']
    response_ratio=round(df.iloc[1]['Responses']/df.iloc[1]['Population'],2)
    learn_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    
    #Assessment and feedback
    positivity=df.iloc[2]['Positivity measure (%)']
    sd=df.iloc[2]['Standard deviation']
    response_ratio=round(df.iloc[2]['Responses']/df.iloc[2]['Population'],2)
    assess_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    #Academic Support
    positivity=df.iloc[3]['Positivity measure (%)']
    sd=df.iloc[3]['Standard deviation']
    response_ratio=round(df.iloc[3]['Responses']/df.iloc[3]['Population'],2)
    support_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    # Organisation and Management
    positivity=df.iloc[4]['Positivity measure (%)']
    sd=df.iloc[4]['Standard deviation']
    response_ratio=round(df.iloc[4]['Responses']/df.iloc[4]['Population'],2)
    organisation_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    # Learning Resources
    positivity=df.iloc[5]['Positivity measure (%)']
    sd=df.iloc[5]['Standard deviation']
    response_ratio=round(df.iloc[5]['Responses']/df.iloc[5]['Population'],2)
    resource_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    # Student Voice
    positivity=df.iloc[6]['Positivity measure (%)']
    sd=df.iloc[6]['Standard deviation']
    response_ratio=round(df.iloc[6]['Responses']/df.iloc[6]['Population'],2)
    voice_dict[name]={'Positivity Measure(%)':positivity, 'Standard Deviation': sd, 'Response Ratio': response_ratio}

    




In [102]:
teach_df=pd.DataFrame(teach_dict)
print('Student Views on Teaching Quality') #a higher percentage means more students think positively of that factor
teach_df.transpose()

Student Views on Teaching Quality


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           86.5                 1.0            0.67
Oxford                        93.1                 0.8            0.50
UCL                           84.4                 0.5            0.72
Birmingham                    84.1                 0.5            0.69
Edinburgh                     85.2                 0.6            0.65
Glasgow                       98.1                 3.4            0.71
Imperial                      89.3                 0.7            0.72
KCL                           83.3                 0.6            0.70
Manchester                    81.7                 0.5            0.74
Norwich                       82.6                 1.5            0.82
Strathclyde                   94.1                 3.9            0.67
Warwick                       87.1                 5.8            0.72

In [103]:
learn_df=pd.DataFrame(learn_dict)
print('Student Views on Learning Opportunities')
learn_df.transpose()

Student Views on Learning Opportunities


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           79.4                 1.1            0.67
Oxford                        82.1                 0.9            0.50
UCL                           79.4                 0.6            0.72
Birmingham                    79.2                 0.6            0.69
Edinburgh                     75.5                 0.7            0.65
Glasgow                       96.2                 4.0            0.71
Imperial                      85.1                 0.8            0.72
KCL                           77.7                 0.6            0.70
Manchester                    76.9                 0.5            0.74
Norwich                       79.1                 1.6            0.82
Strathclyde                   92.0                 4.4            0.67
Warwick                       90.3                 5.9            0.72

In [104]:
print('Student Views on Assessment Quality')
assess_df=pd.DataFrame(assess_dict)
assess_df.transpose()

Student Views on Assessment Quality


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           71.1                 1.2            0.67
Oxford                        74.1                 1.0            0.50
UCL                           68.6                 0.6            0.72
Birmingham                    68.2                 0.7            0.69
Edinburgh                     63.4                 0.8            0.65
Glasgow                       95.6                 3.8            0.71
Imperial                      68.7                 1.0            0.72
KCL                           67.9                 0.7            0.70
Manchester                    67.9                 0.6            0.74
Norwich                       81.6                 1.6            0.82
Strathclyde                   89.3                 4.4            0.67
Warwick                       91.0                 5.5            0.72

In [105]:
print('Student Views on Academic Support Quality')
support_df=pd.DataFrame(support_dict)
support_df.transpose()

Student Views on Academic Support Quality


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           87.4                 1.0            0.67
Oxford                        90.4                 0.8            0.50
UCL                           84.1                 0.5            0.72
Birmingham                    81.4                 0.6            0.69
Edinburgh                     81.3                 0.6            0.65
Glasgow                      100.0                 3.5            0.71
Imperial                      86.6                 0.8            0.72
KCL                           80.6                 0.6            0.70
Manchester                    81.3                 0.5            0.74
Norwich                       87.4                 1.4            0.82
Strathclyde                   97.1                 3.8            0.67
Warwick                       71.0                 7.0            0.72

In [106]:
print('Student Views on Course Organisation Quality')
organisation_df=pd.DataFrame(organisation_dict)
organisation_df.transpose()

Student Views on Course Organisation Quality


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           80.7                 1.1            0.67
Oxford                        67.6                 1.1            0.50
UCL                           74.0                 0.6            0.72
Birmingham                    72.5                 0.7            0.69
Edinburgh                     69.1                 0.8            0.65
Glasgow                       98.1                 3.7            0.71
Imperial                      73.5                 0.9            0.72
KCL                           66.6                 0.7            0.70
Manchester                    70.2                 0.6            0.74
Norwich                       68.7                 1.8            0.82
Strathclyde                   87.3                 4.7            0.67
Warwick                       74.2                 7.0            0.72

In [107]:
print('Student Views on Student Voice')
resource_df=pd.DataFrame(resource_dict)
resource_df.transpose()

Student Views on Student Voice


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           87.1                 0.9            0.67
Oxford                        92.2                 0.8            0.50
UCL                           89.4                 0.5            0.71
Birmingham                    87.2                 0.5            0.69
Edinburgh                     85.7                 0.6            0.65
Glasgow                       94.9                 3.8            0.69
Imperial                      91.2                 0.7            0.72
KCL                           85.1                 0.5            0.70
Manchester                    81.9                 0.5            0.74
Norwich                       85.5                 1.4            0.82
Strathclyde                   92.0                 4.1            0.66
Warwick                       81.7                 6.2            0.72

In [108]:
print('Student opinions on Student Voice')
voice_df=pd.DataFrame(voice_dict)
voice_df.transpose()

Student opinions on Student Voice


Positivity Measure(%)  Standard Deviation  Response Ratio
LSE                           67.2                 1.2            0.67
Oxford                        61.9                 1.1            0.50
UCL                           73.1                 0.6            0.71
Birmingham                    64.4                 0.7            0.69
Edinburgh                     65.6                 0.8            0.65
Glasgow                       90.6                 5.4            0.71
Imperial                      76.7                 0.9            0.72
KCL                           66.5                 0.7            0.70
Manchester                    64.4                 0.6            0.74
Norwich                       72.0                 1.8            0.82
Strathclyde                   77.5                 6.0            0.67
Warwick                       69.9                 8.0            0.72

Another research question involved investigating how student satisfaction varied by degree. Firstly, we see how this works *within* LSE, then compare certain courses across universities.

In [80]:
lse_subjectdat=univ_df['LSE']
lse_subjectdat['Subject'].value_counts(dropna=False)

Subject
Law                                                408
Psychology                                         272
Mathematical sciences                              272
Geography, earth and environmental studies         272
Economics                                          272
Politics                                           272
Business and management                            272
Language and area studies                          136
Human geography                                    136
Philosophy                                         136
Philosophy and religious studies                   136
History                                            136
History and archaeology                            136
Historical, philosophical and religious studies    136
Asian studies                                      136
Languages and area studies                         136
Management studies                                 136
Accounting                                         136
Fi

Many of these subjects overlap and so for the following analysis we will ignore certain categories. For example, we will focus on only Psychology instead of Psychology (non-specific) and both Mathematics and Statistics individually rather than 'Mathematical sciences'. We will also remove the missing values since they are connected to the overall 'rating' across all subjects (the metric previously used to compare universities).

In [83]:
rows_to_drop=[
    "Mathematical sciences",
    "Human geography",
    "Philosophy and religious studies",
    "History and archaeology",
    "Historical, philosophical and religious studies",
    "Languages and area studies",
    "Management studies",
    "Psychology (non-specific)",
    "Sociology, social policy and anthropology",
    "Social sciences"
]


lse_subjectdat=lse_subjectdat[~lse_subjectdat['Subject'].isin(rows_to_drop)]
lse_subjectdat.dropna(subset=['Subject'], inplace=True)
lse_subjectdat['Subject'].value_counts(dropna=False)

Subject
Law                                           408
Economics                                     272
Business and management                       272
Politics                                      272
Psychology                                    272
Geography, earth and environmental studies    272
Anthropology                                  136
Social policy                                 136
Sociology                                     136
Mathematics                                   136
Statistics                                    136
Finance                                       136
Accounting                                    136
Language and area studies                     136
Asian studies                                 136
History                                       136
Philosophy                                    136
Name: count, dtype: int64

We then calculate the specific positivity scores for each subject.

The data differentiates between students with a First and 'all undergraduates', but we could not see any difference between the data collected for both values (i.e. the rows were identical with the one difference being some had 'First') which led us to believe there was no real need to display this distinction.

In [100]:
for name in lse_subjectdat['Subject'].unique():
    
    df=lse_subjectdat[(lse_subjectdat['Subject']==name) & (lse_subjectdat['Level of study']=='All undergraduates')][-7:]
    
    
    


Level of study Subject level Subject code  \
4821   First degree          CAH2     CAH26-01   
4822   First degree          CAH2     CAH26-01   
4823   First degree          CAH2     CAH26-01   
4824   First degree          CAH2     CAH26-01   
4825   First degree          CAH2     CAH26-01   
4826   First degree          CAH2     CAH26-01   
4827   First degree          CAH2     CAH26-01   

                                         Subject  \
4821  Geography, earth and environmental studies   
4822  Geography, earth and environmental studies   
4823  Geography, earth and environmental studies   
4824  Geography, earth and environmental studies   
4825  Geography, earth and environmental studies   
4826  Geography, earth and environmental studies   
4827  Geography, earth and environmental studies   

                                  Question  Responses  Population  \
4821        Theme 1: Teaching on my course       51.4        73.7   
4822       Theme 2: Learning opportunities       51.4        73.7   
4823      Theme 3: Assessment and feedback       51.4        73.7   
4824             Theme 4: Academic support       51.4        73.7   
4825  Theme 5: Organisation and management       51.4        73.7   
4826           Theme 6: Learning resources       51.4        73.7   
4827                Theme 7: Student voice       51.4        73.7   

      Positivity measure (%)  Standard deviation  
4821                    78.6                 5.1  
4822                    75.3                 5.5  
4823                    64.3                 6.2  
4824                    82.8                 4.9  
4825                    74.7                 5.7  
4826                    84.5                 4.6  
4827                    58.9                 6.3

### Reddit LSE data

A refresh of the format of the data frame used to store this data:

In [23]:
lse_reddit_data_df = pd.read_csv("Data/reddit_data.csv")
lse_reddit_data_df.head(5)

Title  Score  \
0                  The irony of LSE being a socialist institution    327   
1                                       Got kicked outta LSE.....    238   
2                           why are masters degrees so expensive?    207   
3  TIL some unis have worse graduate prospects than non-graduates    200   
4                                           University subreddits    190   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Top Comment  
0  Don’t really want to give super identifying details but I’m currently a postgraduate student in LSE and although my programme isn’t under the department of economics, I’m in an economics heavy programme. My professors are quite obviously pro-tax (heavily guided by data and evidence of course), pro-government regulations and interventions in the free market, pro-reform and redistribution through public administration and services, and actively teach us how to see through monopolies, inequalities, and undemocratic tactics administered in western free market countries (primarily the US). I think that we may still be considered centrists by hardcore socialists but we are clearly left wing compared to right wing, conservatives, libertarians, and whatever.  
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Can you share more details? Specifically.\n\nDid you see a DR about such issues and have a record of it?\n\nDid you apply for extenuating circumstances at all?\n\nWas you offered a resit opportunity at all?\n\nWhat is your grounds for appeal, and are you appealing only specifically the final fail decision?  
2                                                                                                                                                                                                                                                                                                                                                                                                           This doesn't really apply to most masters courses if you are a domestic student, the MFin looks aimed at people who already have professional experience which makes it quite different to most other masters. Even at Cambridge  most course are around £12k so almost coverable with the student loan, but still you do need to find £10k+ for living expenses from somewhere  
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

The first thing to check is the number of duplicate or null values. As you can see below, there is neither for both.
- There are no null values since the code used to access the top comment from each post was written to only add the post to the dataframe if it contained a comment. 

- There are no duplicate posts since the chance of two identical strings of words is highly unlikely.

In [17]:
num_duplicates = lse_reddit_data_df.duplicated().sum()
print(f"Number of duplicate rows: {num_duplicates}")

Number of duplicate rows: 0


In [18]:
total_nulls = lse_reddit_data_df.isnull().sum().sum()
print(f"Number of null values: {total_nulls}")
print(lse_reddit_data_df.shape)

Number of null values: 0
(30, 3)


We only want to consider posts that have a high amount of upvotes, since they are more likely to be helpful and insightful information if other users liked them. We can acheive this by removing posts whose score is below '30', of which there way only one. This number was decided given the maximum score was 327 and the mean score was 97.

It is worth noting that the posts are already sorted in order of descending score, since that is a paramter included in the Reddit API call.

In [19]:
print(f"Maximum score: {lse_reddit_data_df['Score'].max()}")
print(f"Mean score: {int(lse_reddit_data_df['Score'].mean())}")
print(f"Current number of posts: {lse_reddit_data_df.shape[0]}")
num_low_score_posts = (lse_reddit_data_df['Score']<= 30).sum()
print(f"Number of posts with a score of 5 or below: {num_low_score_posts}")
lse_reddit_data_df = lse_reddit_data_df[lse_reddit_data_df['Score'] > 30]
print(f"New number of posts: {lse_reddit_data_df.shape[0]}")


lse_reddit_data_df.head(5)

Maximum score: 327
Mean score: 97
Current number of posts: 30
Number of posts with a score of 5 or below: 1
New number of posts: 29


Title  Score  \
0                  The irony of LSE being a socialist institution    327   
1                                       Got kicked outta LSE.....    238   
2                           why are masters degrees so expensive?    207   
3  TIL some unis have worse graduate prospects than non-graduates    200   
4                                           University subreddits    190   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Top Comment  
0  Don’t really want to give super identifying details but I’m currently a postgraduate student in LSE and although my programme isn’t under the department of economics, I’m in an economics heavy programme. My professors are quite obviously pro-tax (heavily guided by data and evidence of course), pro-government regulations and interventions in the free market, pro-reform and redistribution through public administration and services, and actively teach us how to see through monopolies, inequalities, and undemocratic tactics administered in western free market countries (primarily the US). I think that we may still be considered centrists by hardcore socialists but we are clearly left wing compared to right wing, conservatives, libertarians, and whatever.  
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Can you share more details? Specifically.\n\nDid you see a DR about such issues and have a record of it?\n\nDid you apply for extenuating circumstances at all?\n\nWas you offered a resit opportunity at all?\n\nWhat is your grounds for appeal, and are you appealing only specifically the final fail decision?  
2                                                                                                                                                                                                                                                                                                                                                                                                           This doesn't really apply to most masters courses if you are a domestic student, the MFin looks aimed at people who already have professional experience which makes it quite different to most other masters. Even at Cambridge  most course are around £12k so almost coverable with the student loan, but still you do need to find £10k+ for living expenses from somewhere  
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

### Reddit COVID-19 Data

Similar to the procedure above, the below dataframe about reddit posts that contain the word 'covid' is imported from a csv file and saved as a dataframe.

In [36]:
covid_reddit_data_df = pd.read_csv("Data/covid_reddit_data.csv")
covid_reddit_data_df.head(5)

Title  \
0  Is anyone completely disengaged with university to the point of not attending any lectures/tutorials and only looking at that when it's needed for assignments/exams? I cannot focus with online learning at all and don't even feel like a student   
1                                                                                                                                                                                     Apparently alot of "covid" student have been dropping out of Uni   
2                                                                                                                                                                                                        Important: For people starting uni this year.   
3                                                                                                                                                                                                                                   who'd had thunk it   
4                                                                                                                                                                           Has anyone done a degree they now realise has absolutely no job prospects?   

   Score  \
0    461   
1    389   
2    383   
3    376   
4    361   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Top Comment  
0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [deleted]  
1  I'm an admissions tutor and lecturer. I teach a fairly challenging module but it's actually quite popular and the final average was about 65%. During Covid we went to online assessment, which resulted in the average shooting up. We returned to a proctored exam last year, and the the average was 38%. This was the 2021 intake (its a Year 2 module). A lot have failed other modules so will be deregistered. Some have quit before they are pushed. Its absolutely not their fault and it's really sad. Their A-levels were severely interrupted, they never learnt how to study properly, and they were totally demoralised. Unfortunately, I think the government will just forget about them. I worry they'll become a lost cohort like the Japanese kids who graduated during the Pacific financial crisis.  
2                                                                                                                                                                                                                                                                                

To clean the data frame, rows where the value in the 'Top Comment' column was '[deleted]' are removed as this was considered a null value, and where the score is 0.


In [45]:
#remove null values
covid_reddit_data_df = covid_reddit_data_df[covid_reddit_data_df['Top Comment'] != '[deleted]']

#find summary statistics, remove low  scores
print(f"Maximum score: {covid_reddit_data_df['Score'].max()}")
print(f"Mean score: {int(covid_reddit_data_df['Score'].mean())}")
print(f"Current number of posts: {covid_reddit_data_df.shape[0]}")
num_low_score_posts = (covid_reddit_data_df['Score']== 0).sum()
print(f"Number of posts with a score of 0: {num_low_score_posts}")
covid_reddit_data_df = covid_reddit_data_df[covid_reddit_data_df['Score'] > 0]
print(f"New number of posts: {covid_reddit_data_df.shape[0]}")

#display the full comment in the 'Top Comment' column
pd.set_option('display.max_colwidth', None)

covid_reddit_data_df.head(5)

Maximum score: 389
Mean score: 117
Current number of posts: 94
Number of posts with a score of 0: 0
New number of posts: 94


Title  \
1            Apparently alot of "covid" student have been dropping out of Uni   
2                               Important: For people starting uni this year.   
3                                                          who'd had thunk it   
4  Has anyone done a degree they now realise has absolutely no job prospects?   
5            I wasted my fucking time at uni and now I regret ever even going   

   Score  \
1    389   
2    383   
3    376   
4    361   
5    343   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Top Comment  
1  I'm an admissions tutor and lecturer. I teach a fairly challenging module but it's actually quite popular and the final average was about 65%. During Covid we went to online assessment, which resulted in the average shooting up. We returned to a proctored exam last year, and the the average was 38%. This was the 2021 intake (its a Year 2 module). A lot have failed other modules so will be deregistered. Some have quit before they are pushed. Its absolutely not their fault and it's really sad. Their A-levels were severely interrupted, they never learnt how to study properly, and they were totally demoralised. Unfortunately, I think the government will just forget about them. I worry they'll become a lost cohort like the Japanese kids who graduated during the Pacific financial crisis.  
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Also, not just for you but fellow students, we appreciate there being fewer people to spread whatever arrives on campus.  
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Some of us also seem to make this face upon discovering hangovers.  
4                                                                                                                                                                                                                                                                                                                               

The next step was to extract any information about how COVID-19 impacted universtity students in London. The results were as follows:

Only 3/200 posts mentioned the words in the 'keywords' list below that were linked to the city of London, or the names of the biggest universities in London. Of the three posts outputted, only one details the impact of COVID-19 on student life by discussing online learning in the King's College London Physics degree. However, they are descirbing the benefits of this course due to it having compulsory in-person labs, whereas most posts in the previous dataframe have a negative view.

The scores of these posts are also all lower than the mean score of 117, signifying there are not as relevant compared to other posts about COVID-19.

The lack of results could be explained by the following reasons:

- Whether a university was located in London had no impact on the student satisfaction during/after COVID-19.
- Students chose not to mentioned the university they attended when posting on the website.
- The information was not accurately gathered as there are too many variations of university names or other
identifying phrases of a student being located in London.
- The sample size was to small.



In [46]:
keywords = [' lse ', ' ucl ', 'london', ' kcl ', 'kings', ' icl ', 'imperical',
            'imperical', ' ual ', ' uol ', 'university of london ', 'university of west london',
            'university of east london', 'london metropolitan university', ' london met', 'university of greenwhich',
            'university of westminster', 'middlesex university', 'london southbank',
            "regent's university", 'university of roehampton', 'ravensbourne university', 'royal college of art',' rcoa ',
            'birkbeck', ' bpp university', 'guildhall school', 'royal college of music', ' lbs ', 'london business school',
            'rada']

#find posts containing any keywords
london_filtered_df = covid_reddit_data_df[covid_reddit_data_df['Title'].str.contains('|'.join(keywords), case=False, na=False) |
                 covid_reddit_data_df['Top Comment'].str.contains('|'.join(keywords), case=False, na=False)]

london_filtered_df

Title  \
48  British but being charged international fees because I spent 4 years studying in the US   
62                                                        Questions about clubbing - London   
71                                                            Terrified Of Online Learning?   

    Score  \
48    100   
62     68   
71     51   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Top Comment  
48  I’d say the London university is right. To be considered a home student, you need to be ordinarily resident in the U.K. for the three years before your course starts (e.g. https://www.lse.ac.uk/study-at-lse/international-students/fee-status-classification), which unfortunately you haven’t been, whatever the reason. I’d request a fee status reassessment from both of them to be honest - you wouldn’t want to get half way through the degree at the Scottish university and suddenly have to pay international fees.  
62                                                                                                                                                                                         Pryzm is a famously bad club, terrible atmosphere with awful clientele and a hotbed for spiking. What club you want depends on what sort of music you’re into, I club a lot in London but since I love electronic music it’s much more to go to listen to music rather than to try get off with strangers like the likes of Pryzm are for  
71                                    It depends what your degree is. Also whether you’re going to be in England or Scotland or NI or Wales because rules can be a bit different. If you’re doing a STEM degree, it’s very likely that you will be on campus (likely some online lectures too) - I received word from KCL that STEM degrees like Physics will have to be on campus because of compulsory labs. If you’re doing something like History or Literature, it will be even more likely you’ll be doing most things online.